pkg/reconciler/votereconciler/redis_delete.go

```diff
@@ -27,7 +27,7 @@ func (r *Redis) DeleteVote(inp []*votestorage.Object) ([]*votestorage.Object, er
 
 		// Votes can only be deleted as long as they have not been confirmed
 		// onchain.
-		if !cla.Lifecycle.Pending() {
+		if !inp[i].Lifecycle.Pending() {
```

Comment:
The resource that requires the pending check is the vote that we try to delete. The bug here was that we checked for the lifecycle of the claim that the vote was cast on.